src/alltrue_guardrails/event/loop.py

```
@@ -31,65 +34,61 @@ def __init__(
         Similar to `asyncio.ensure_future` to run the given coroutine in the background whenever the loop is available.
         """
-        task = self._loop.create_task(coroutine)
+        feature = asyncio.run_coroutine_threadsafe(
+            coroutine,
+            loop=self._loop,
+        )
+        setattr(
+            feature,
+            "get_name",
+            lambda: f"Task[{coroutine.__qualname__}]-{id(coroutine)}",
+        )
```

Ruff (0.12.2)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)